source/particle/interpolator/bilinear_least_squares.cc

```diff
@@ -89,14 +90,17 @@ namespace aspect
 
         // Noticed that the size of matrix A is n_particles x matrix_dimension
         // which usually is not a square matrix. Therefore, we find the
-        // least squares solution of Ax=r by solving the "normal" equations
-        // (A^TA) x = A^Tr.
+        // least squares solution of Ax=r by solving reduced QR factorization
+        // Ax = QRc = b -> Q^TQRc = Rc =Q^Tb
+        dealii::ImplicitQR<dealii::Vector<double>> qr;
         const unsigned int matrix_dimension = (dim == 2) ? 4 : 8;
-        dealii::LAPACKFullMatrix<double> A(n_particles, matrix_dimension);
-        std::vector<Vector<double>> r(n_particle_properties, Vector<double>(n_particles));
+        std::vector<dealii::Vector<double>> A(matrix_dimension, dealii::Vector<double>(n_particles));
```

Comment:
Please leave a comment on this line why you store the matrix A as a vector of vectors instead of a matrix (I suppose it is because the ImplicitQR requires you to add it column by column to the QR factorization?).

Reply:
I believe that is correct.  @sac-bsa will know for sure.

Reply:
Yes, that reasoning is correct, I will add a comment here.